File: frontend/cli/src/app/pedidos/interfaces/planificacion.ts
import { Equipo } from "../../talleres/interfaces/equipo";
import { Tarea } from "../../productos/interfaces/tarea";
import { OrdenTrabajo } from "./ordenTrabajo";

export interface Planificacion {
    id: number;
    inicio: Date;
    fin: Date;
    equipo: Equipo;
    tarea: Tarea;
    ordenTrabajo: OrdenTrabajo;
}

File: backend/src/main/java/unpsjb/labprog/backend/model/Planificacion.java
package unpsjb.labprog.backend.model;

import java.sql.Timestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Getter
@Setter
@Entity
public class Planificacion {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @ManyToOne
    private Tarea tarea;

    @ManyToOne
    private PedidoFabricacion pedido;

    @ManyToOne
    private Equipo equipo;

    @Column(nullable = false)
    private Timestamp inicio;

    @Column(nullable = false)
    private Timestamp fin;
}
